services/src/test/scala/cromwell/pipeline/service/GitLabProjectVersioningTest.scala

```diff
@@ -96,6 +96,54 @@ class GitLabProjectVersioningTest
       }
     }
 
+    "getUpdatedProjectVersion" should {
+      val userVersion = PipelineVersion("v0.0.1")
+      val project = TestProjectUtils.getDummyProject()
+      def request(project: Project): Future[Response[Seq[GitLabVersion]]] =
+        mockHttpClient.get[Seq[GitLabVersion]](
+          url = exact(gitLabConfig.url + "projects/" + project.repositoryId.value + "/repository/tags"),
+          headers = any[Map[String, String]],
+          params = any[Map[String, String]]
+        )(any[ExecutionContext], any[Reads[Seq[GitLabVersion]]])
+
+      "return correct PipelineVersion" taggedAs Service in {
+
+        when(request(project)).thenReturn {
+          Future.successful {
+            Response[Seq[GitLabVersion]](
+              HttpStatusCodes.OK,
+              SuccessResponseBody[Seq[GitLabVersion]](dummyVersions),
+              EmptyHeaders
+            )
+          }
+        }
+
+        gitLabProjectVersioning.getUpdatedProjectVersion(project, Some(userVersion)).map(_ shouldBe Right(userVersion))
+      }
+
+      "throw VersioningException.ProjectException" taggedAs Service in {
```

Comment:
```suggestion
      "fail with VersioningException.ProjectException" taggedAs Service in {
```
Throwing an exception and failing with some result are two different things

Reply:
Thanks for the clarification